tomcat/src/main/java/org/apache/http/HttpRequest.java

```diff
@@ -0,0 +1,71 @@
+package org.apache.http;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.exception.InvalidRequestException;
+
+public class HttpRequest {
+
+    private final HttpMethod method;
+    private final String uri;
+    private final Map<String, String> queryString;
+    private final HttpVersion version;
+
+    public HttpRequest(List<String> message) {
+        if (message.size() <= 0) {
+            throw new InvalidRequestException("요청 메세지가 비어있습니다.");
+        }
+        List<String> startLine = List.of(message.getFirst().split(" "));
+        if (startLine.size() < 3) {
+            throw new InvalidRequestException("요청 메세지의 시작라인 형식이 올바르지 않습니다.");
+        }
+        method = HttpMethod.valueOf(startLine.get(0));
+        uri = parseUri(startLine.get(1));
+        queryString = parseQueryParam(startLine.get(1));
+        version = HttpVersion.parse(startLine.get(2));
+    }
+
+    public boolean checkQueryStringExistence(String key) {
+        return queryString.containsKey(key);
+    }
+
+    public HttpMethod getMethod() {
+        return method;
+    }
+
+    public String getUri() {
+        return uri;
+    }
+
+    public HttpVersion getVersion() {
+        return version;
+    }
+
+    public String getQueryString(String key) {
+        return queryString.get(key);
+    }
+
+    private String parseUri(String uriLine) {
+        List<String> startLinePart = List.of(uriLine.split("\\?"));
+        return startLinePart.getFirst();
+    }
+
+    private Map<String, String> parseQueryParam(String uriLine) {
+        HashMap<String, String> queryStrings = new HashMap<>();
+
+        List<String> startLinePart = List.of(uriLine.split("\\?"));
+        if (startLinePart.size() < 2) {
+            return queryStrings;
+        }
```

Comment:
개인적으로 2가 어떤 의미인지 파악하기 어려웠어요. 이런 의미겠죠?
?로 나누었을 때, 2개 미만으로 존재함 -> 쿼리 스트링이 없음
아래 방식은 어떤가요?
```suggestion
        List<String> startLinePart = List.of(uriLine.split("\\?"));
        if (!uriLine.contains("?")) {
            return queryStrings;
        }
```
? 가 포함되지 않았음을 명시적으로 전달 가능해서 가독성이 좋을 것 같아요!

if 문을 List<String> startLinePart = List.of(uriLine.split("\\?")); 위로 옮긴다면, 더 좋겠네요~

Reply:
아마가 말씀해주신대로 "!uriLine.contains("?")" 형태가 가독성이 훨씬 높은 것 같습니다! 약간의 리펙토링을 추가해서 바로 수정했습니다!